Tests/SwiftSyntaxMacroExpansionTest/MacroExpansionContextTests.swift

```diff
@@ -0,0 +1,73 @@
+  //===----------------------------------------------------------------------===//
+  //
+  // This source file is part of the Swift.org open source project
+  //
+  // Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+  // Licensed under Apache License v2.0 with Runtime Library Exception
+  //
+  // See https://swift.org/LICENSE.txt for license information
+  // See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+  //
+  //===----------------------------------------------------------------------===//
+
+  //==========================================================================//
+  // IMPORTANT: The macros defined in this file are intended to test the      //
+  // behavior of MacroSystem. Many of them do not serve as good examples of   //
+  // how macros should be written. In particular, they often lack error       //
+  // handling because it is not needed in the few test cases in which these   //
+  // macros are invoked.                                                      //
+  //==========================================================================//
+
+import SwiftSyntax
+import SwiftDiagnostics
+@_spi(Diagnostics) import SwiftSyntaxBuilder
+import SwiftSyntaxMacros
+import SwiftSyntaxMacroExpansion
+import SwiftSyntaxMacrosTestSupport
+import XCTest
+
+//private class TestMacroExpansionContext: MacroExpansionContext {
+//  public var diagnostics = [Diagnostic]()
+//
+//  func diagnose(_ diagnostic: Diagnostic) {
+//    diagnostics.append(diagnostic)
+//  }
+//
+//  func makeUniqueName(_ name: String) -> TokenSyntax {
+//    return TokenSyntax.identifier(name)
+//  }
+//
+//  func location(
+//    of node: some SyntaxProtocol,
+//    at position: PositionInSyntaxNode,
+//    filePathMode: SourceLocationFilePathMode
+//  ) -> AbstractSourceLocation? {
+//    return nil
+//  }
+//}
+
+private struct DummyError: Error {
+  static let diagnosticTestError = DummyError()
+}
+
+final class MacroExpansionContextTests: XCTestCase {
+
+  func testMacroExpansionContextAddDiagnosticsAddsSwiftSyntaxInterpolationErrorsWithWrappingMessage(){
+    let context = BasicMacroExpansionContext()
+    let error = SyntaxStringInterpolationError.producedInvalidNodeType(expectedType: DeclSyntax.self, actualType: ExprSyntax.self)
+    // Since we only care about the error switch inside of addDagnostics, we don't care about the particular node we're passing in.
+    context.addDiagnostics(from: error, node: ExprSyntax("1"))
+
+    XCTAssertEqual(context.diagnostics.count, 1)
+    XCTAssertTrue(context.diagnostics.first!.message.starts(with: "Internal macro error:"))
```

Comment:
I think you should be able to test the real-case scenario when this happens by defining an extension macro that returns 

```swift
try ExtensionDeclSyntax("var x: Int")
```

And then use `assertMacroExpansion` to check that the emitted error contains `Internal macro error:`